releasenotes/notes/file-tailing-wildcard-match-mode-73f58cbf7b2a0f73.yaml

```diff
@@ -0,0 +1,17 @@
+# Each section from every release note are combined when the
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    A new config option ``logs_config.file_wildcard_selection_mode`` has been added.
+    This option allows users to configure how log wildcard file matches are
+    prioritized if the number of matches exceeds ``logs_config.open_files_limit``
+
+    This defaults to ``by_name`` which is the current behavior.
+    The new option is ``by_modification_time`` which will prioritize more recently
+    modified files, but it comes with a performance penalty compared to ``by_name``.
```

Comment:
```suggestion
    A new config option, ``logs_config.file_wildcard_selection_mode``, 
    allows you to configure how log wildcard file matches are
    prioritized if the number of matches exceeds ``logs_config.open_files_limit``.

    The option defaults to ``by_name`` which is the previous behavior.
    The new option is ``by_modification_time`` which prioritizes more recently
    modified files, but using it can result in slower performance compared to using ``by_name``.
```